## STORYTELLING CASE STUDY : Airbnb, NYC

# Introduction:
This assignment aims to give an idea of applying Data Storytelling concept in a real business scenario,In this assignment, will develop a basic understanding of some important insights based on various attributes in the dataset so as to increase 
 the revenue.

### Problem background:

Suppose that you are working as a data analyst at Airbnb.For the past few months, Airbnb has seen a major decline in revenue.- Now that the restrictions have started lifting and people have started to travel more, Airbnb wants to make sure that it is 
fully prepared for this change.

### Business Understanding:

The different leaders at Airbnb want to understand some important insights based on various attributes in the dataset so as 
to increase the revenue such as -
Which type of hosts to acquire more and where?

The categorisation of customers based on their preferences.

What are the neighbourhoods they need to target?

What is the pricing ranges preferred by customers?

The various kinds of properties that exist w.r.t. customer preferences.

Adjustments in the existing properties to make it more customer-oriented.

What are the most popular localities and properties in New York currently?

How to get unpopular properties more traction?

#### Business Objectives:

To prepare for the next best steps that Airbnb needs to take as a business, you have been asked to analyse a dataset consisting of various Airbnb listings in New York.

### Presentation - I
Data Analysis Managers: These people manage the data analysts directly for processes and their technical expertise is basic.

Lead Data Analyst: The lead data analyst looks after the entire team of data and business analysts and is technically sound.

### Presentation - II
Head of Acquisitions and Operations, NYC: This head looks after all the property and host acquisitions and operations. Acquisition of the best properties, price negotiation, and negotiating the services the properties offer falls under the purview of this role.

Head of User Experience, NYC: The head of user experience looks after the customer preferences and also handles the properties listed on the website and the Airbnb app. Basically, the head of user experience tries to optimise the order of property listing in certain neighbourhoods and cities in order to get every property the optimal amount of traction.

### Import the necessary libraries

In [41]:
#import the warnings.
import warnings
warnings.filterwarnings(action='ignore')

In [42]:
#import the useful libraries.
import warnings
warnings.filterwarnings("ignore")
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [43]:
#Display all columns and rows of dataframe
pd.set_option("display.max_rows",None)
pd.set_option("display.max_columns",None)

#### Read and make undersatnding of the Data set. 

In [44]:
#Read the data set of "application_data" in df.
airbnb = pd.read_csv(".\\AB_NYC_2019.csv")

In [45]:
# Print the shape of the data frame.
airbnb.shape

(48895, 16)

In [46]:
#Print the head of the data frame.
airbnb.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  19-10-2018   
1  Entire home/apt    225               1                 45  21-05-2019   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  05-07-2019   
4  Entire home/apt     80              10                  9  19-11-2018   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [47]:
#describe the data frame
airbnb.describe()

id       host_id      latitude     longitude         price  \
count  4.889500e+04  4.889500e+04  48895.000000  48895.000000  48895.000000   
mean   1.901714e+07  6.762001e+07     40.728949    -73.952170    152.720687   
std    1.098311e+07  7.861097e+07      0.054530      0.046157    240.154170   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420      0.000000   
25%    9.471945e+06  7.822033e+06     40.690100    -73.983070     69.000000   
50%    1.967728e+07  3.079382e+07     40.723070    -73.955680    106.000000   
75%    2.915218e+07  1.074344e+08     40.763115    -73.936275    175.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48895.000000       48895.000000       38843.000000   
mean         7.029962          23.274466           1.373221   
std         20.510550          44.550582           1.680442   
min          1.000000           0.000000           0.010000   
25%          1.000000           1.000000           0.190000   
50%          3.000000           5.000000           0.720000   
75%          5.000000          24.000000           2.020000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48895.000000      48895.000000  
mean                         7.143982        112.781327  
std                         32.952519        131.622289  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

In [48]:
#print the information of variables to check their data types.
airbnb.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

#### Data cleanup 

In [49]:
# Percentage of missing values in every column
Null_percentage=(100*airbnb.isnull().mean().sort_values(ascending=False))
Null_percentage

last_review                       20.558339
reviews_per_month                 20.558339
host_name                          0.042949
name                               0.032723
id                                 0.000000
host_id                            0.000000
neighbourhood_group                0.000000
neighbourhood                      0.000000
latitude                           0.000000
longitude                          0.000000
room_type                          0.000000
price                              0.000000
minimum_nights                     0.000000
number_of_reviews                  0.000000
calculated_host_listings_count     0.000000
availability_365                   0.000000
dtype: float64

In [50]:
# Calculating the missing values in the dataset
airbnb.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [51]:
# Now we have the missing values, there are certain columns that are not efficient to the dataset
airbnb.drop(['id','name','last_review'], axis = 1, inplace = True)

In [52]:
# View whether the columns are dropped
airbnb.head(5)

host_id    host_name neighbourhood_group neighbourhood  latitude  \
0     2787         John            Brooklyn    Kensington  40.64749   
1     2845     Jennifer           Manhattan       Midtown  40.75362   
2     4632    Elisabeth           Manhattan        Harlem  40.80902   
3     4869  LisaRoxanne            Brooklyn  Clinton Hill  40.68514   
4     7192        Laura           Manhattan   East Harlem  40.79851   

   longitude        room_type  price  minimum_nights  number_of_reviews  \
0  -73.97237     Private room    149               1                  9   
1  -73.98377  Entire home/apt    225               1                 45   
2  -73.94190     Private room    150               3                  0   
3  -73.95976  Entire home/apt     89               1                270   
4  -73.94399  Entire home/apt     80              10                  9   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [53]:
# Check the rows and columns of the dataset
airbnb.shape

(48895, 13)

The dataset contains 48895 rows and 13 columns

Now we have to check whether there are any missing values in the dataset

In [54]:
# Now reviews per month contains more missing values which should be replaced with 0 respectively
airbnb.fillna({'reviews_per_month':0},inplace=True)

In [55]:
airbnb.reviews_per_month.isnull().sum()

0

In [56]:
len(airbnb.room_type.unique())

3

In [57]:
airbnb.neighbourhood_group.unique()

array(['Brooklyn', 'Manhattan', 'Queens', 'Staten Island', 'Bronx'],
      dtype=object)

In [58]:
len(airbnb.neighbourhood.unique())

221

In [59]:
airbnb.to_csv(r'.\\airbnb_dataset.csv',index=False, header=True)

In [60]:
airbnb.host_id.value_counts().head(10)

host_id
219517861    327
107434423    232
30283594     121
137358866    103
16098958      96
12243051      96
61391963      91
22541573      87
200380610     65
7503643       52
Name: count, dtype: int64

In [61]:
airbnb2 = airbnb.sort_values(by="calculated_host_listings_count",ascending=False)
airbnb2.head()

host_id     host_name neighbourhood_group       neighbourhood  \
39773  219517861  Sonder (NYC)           Manhattan      Hell's Kitchen   
41463  219517861  Sonder (NYC)           Manhattan  Financial District   
41469  219517861  Sonder (NYC)           Manhattan  Financial District   
38294  219517861  Sonder (NYC)           Manhattan  Financial District   
41468  219517861  Sonder (NYC)           Manhattan  Financial District   

       latitude  longitude        room_type  price  minimum_nights  \
39773  40.76037  -73.99744  Entire home/apt    185              29   
41463  40.70782  -74.01227  Entire home/apt    396               2   
41469  40.70620  -74.01192  Entire home/apt    498               2   
38294  40.70771  -74.00641  Entire home/apt    229              29   
41468  40.70726  -74.01060  Entire home/apt    229               2   

       number_of_reviews  reviews_per_month  calculated_host_listings_count  \
39773                  1               1.00                             327   
41463                  8               2.12                             327   
41469                  8               2.50                             327   
38294                  1               0.73                             327   
41468                  2               0.77                             327   

       availability_365  
39773               332  
41463               289  
41469               255  
38294               219  
41468               351